### PR TITLE
Bug 1169263 - Switch from Vagrant to docker-compose

### PR DESCRIPTION
**Pull request by armenzg**
Rebased Ed's WIP #4775

### CODE REVIEW

**Comment by armenzg** [comment]
@djmitche Hi Dustin, I'm currently greening up the Selenium tests, however, this PR should give you an idea of where I'm standing at. `docker-compose up` gets the project going and you can load http://localhost:5000 which talks to the running backend. If I could get some feedback before EOW I should be able to handle it before I head out for 3 weeks of parental leave.

Let me know if you would like to jump over Vidyo to go over the PR together.

**Comment by armenzg** [comment]
@djmitche let me know if you believe having a development image and a test image is warranted. I wanted to keep the development image a bit smaller.

That will remove having the 'testing' entry in docker-compose.yml

File: docs/installation.md [hunk 1]
```diff
@@ -92,92 +84,51 @@ The tests will perform an initial run and then re-execute each time a project fi
...
 
-- Install [Virtualbox] and [Vagrant] (latest versions recommended).
-- Linux only: An nfsd server is required. You can install this on Ubuntu by running `apt-get install nfs-common nfs-kernel-server`
+- Install Docker & docker-compose (both are installed if using Docker for Windows/Mac).
 
-### Setting up Vagrant
+- If you just wish to [run the tests](common_tasks.md#running-the-tests),
+  you can stop now without performing the remaining steps.
+
+### Starting a local Treeherder instance
```
Comment (sarah-clements) [inline docs/installation.md]: I'm not seeing a note about running the backend with a preset environment variable - `DATABASE_URL`-  and then what command to use to run the frontend.

**Comment by armenzg** [comment]
> I tried this out and it looks good to me, but camd and Ionut should test it before we merge. One question though - can you set the 5000 and 8000 ports as defaults so you don't need to specify `-p 8000:8000` when running backend?

When running docker-compose with anything but the default call (`docker-compose up`) it decides that you want to run something custom.

For instance if I do the default call `docker-compose up` I get the ports being defined:
image

If I run `docker-compose run backend bash` (I just append a command different than the default) I will see that the ports are not forwarded:
image

If I add `EXPOSE 8000` to the `dev.Dockerfile` and I run `docker-compose run backend` I will see an improvement:
image

What I would want to see is `0.0.0.0:8000->8000/tcp` instead of `8000/tcp`.

I read this in the documentation:

> The EXPOSE instruction does not actually publish the port. It functions as a type of documentation between the person who builds the image and the person who runs the container, about which ports are intended to be published. To actually publish the port when running the container, use the -p flag on docker run to publish and map one or more ports, or the -P flag to publish all exposed ports and map them to high-order ports

@djmitche any suggestions?

**Comment by djmitche** [comment]
I only learned about `docker-compose` from this PR, so I think I'm not much help here.

**Comment by edmorley** [comment]
> If I run `docker-compose run backend bash` (I just append a command different than the default) I will see that the ports are not forwarded:

When using `docker-compose run` it doesn't expose ports by default - to override use `--service-ports` - see:
https://docs.docker.com/compose/reference/run/

(And hi! 👋 )

File: docs/installation.md [hunk 2]
```diff
@@ -92,92 +84,51 @@ The tests will perform an initial run and then re-execute each time a project fi
 
 To get started:
 
-- Install [Virtualbox] and [Vagrant] (latest versions recommended).
-- Linux only: An nfsd server is required. You can install this on Ubuntu by running `apt-get install nfs-common nfs-kernel-server`
+- Install Docker & docker-compose (both are installed if using Docker for Windows/Mac).
 
-### Setting up Vagrant
+- If you just wish to [run the tests](common_tasks.md#running-the-tests),
```
[the remark this answers is not in the document]
Comment (camd) [reply docs/installation.md]: I fixed this myself

**Comment by camd** [comment]
I added back in the ignore for ``.vagrant`` in ``.gitignore`` so that if people have vagrant files lying around, they won't accidentally get added.

**Pull request merged**